zerver/lib/upload.py

```diff
@@ -33,10 +35,8 @@
 # slugify.
 
 def sanitize_name(name):
-    split_name = name.split('.')
-    base = ".".join(split_name[:-1])
-    extension = split_name[-1]
-    return slugify(base) + "." + slugify(extension)
+    fileName = force_text(name).strip().replace(' ', '_')
+    return re.sub(r'(?u)[^-\w.]', '', fileName)
```

Comment:
It looks like this is a simpler version of slugify() except with:
* No support for the allow_unicode option (which seems fine)
* slightly different semantics.

See https://docs.djangoproject.com/en/1.9/_modules/django/utils/text/#slugify for the slugify implementation.

I think I'd be a lot more comfortable with something that is more clearly based on slugify, e.g. copy in the code of slugify, delete the allow_unicode option support, and change the regexp to allow '.' characters and skip the lower-casing.  I think this function also needs to document when we believe it is correct in a comment, explaining how it was generated, since it is security-sensitive code and that explanation will make it much more clear in what way(s) it is safe to modify it in the future.  

Does that seem reasonable, @varshit97 ?